test/common/alerts_exist.go

```diff
@@ -0,0 +1,530 @@
+			"UnifiedPushJavaGCTimePerMinuteScavenge",
+			"UnifiedPushJavaDeadlockedThreads",
+			"UnifiedPushMessagesFailures",
+		},
+	},
+	{
+		File: "redhat-rhmi-ups-operator-unifiedpush-operator.yaml",
+		Rules: []string{
+			"UnifiedPushOperatorDown",
+		},
+	},
+}
+
+var expectedAWSRules = []alertsTestRule{
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-threescale-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-threescale-backend-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-threescale-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"3scalePostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-threescale-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-threescale-backend-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-threescale-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"3scalePostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-ups-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"upsPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-ups-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"upsPostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-codeready-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"codeready-workspacesPostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-codeready-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"codeready-workspacesPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-rhssouser-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"user-ssoPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-rhssouser-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"user-ssoPostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-rhsso-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"rhssoPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-rhsso-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"rhssoPostgresInstanceUnavailable",
+		},
+	},
+}
+
+func TestIntegreatlyAlertsExist(t *testing.T, ctx *TestingContext) {
+	// get the RHMI custom resource to check what storage type is being used
+	rhmi := &v1alpha1.RHMI{}
+	ns := fmt.Sprintf("%soperator", namespacePrefix)
+	err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: installationName, Namespace: ns}, rhmi)
+	if err != nil {
+		t.Fatal("error getting RHMI CR:", err)
+	}
```

Comment:
This information should be a test parameter or flag (don't know what can be used in golang), and this would be a problem of many other tests so it should be solved globally

ping @wei-lee

Reply:
@b1zzu can be elaborate on what exactly do you mean here?

Reply:
The code here is checking if the cluster is using or not AWS resources, but when we run the test pipeline we already know if the cluster we have provisioned should use or not the aws resources, therefore we should pass this information to the tests instant of the tests finding it out

Reply:
I think checking the CR is valid. It's the system of record for what type of storage is being used.
No need to double up on this. The test would instruct what to set to the CR to, not the other way around.